azure-pipelines.yml

```diff
@@ -128,8 +128,16 @@ variables:
   NEED_XMLSTARLET: $(needXmlstarlet)
   NEED_MDL: $(needMdl)
   BUILD_REASON: $[variables['Build.Reason']]
+  M2_HOME: '/usr/local/apache-maven-3.8.6'
+  PATH: '/usr/local/apache-maven-3.8.6/bin:$(PATH)'
 
 steps:
+  - bash: |
+      wget https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.6/apache-maven-3.8.6-bin.tar.gz
+      tar xzvf apache-maven-3.8.6-bin.tar.gz
```

Comment:
lets use https://stackoverflow.com/a/52343639

Reply:
No need to update azure, it has latest - 3.8.8
https://github.com/checkstyle/checkstyle/pull/14078#issuecomment-1832825993
We are ok to run mvnw that is above min required.